src/services/ExposureNotificationService/ExposureNotificationService.ts

```diff
@@ -411,20 +396,16 @@ export class ExposureNotificationService {
     return this.finalize();
   }
 
-  private async getKeysFileUrls() {
+  private async getKeysFileUrls(): Promise<any> {
     const currentStatus = this.exposureStatus.get();
     const keysFileUrls: string[] = [];
-    const generator = this.keysSinceLastFetch(currentStatus.lastChecked?.period);
-    captureMessage('currentStatus.lastChecked?.period', {period: currentStatus.lastChecked?.period});
     let lastCheckedPeriod = currentStatus.lastChecked?.period;
-    while (true) {
-      const {value, done} = await generator.next();
-      if (done) break;
-      if (!value) continue;
-      const {keysFileUrl, period} = value;
-      captureMessage('loop period', {period});
+    const periodsSinceLastFetch = this.getPeriodsSinceLastFetch(lastCheckedPeriod);
+    for (const period of periodsSinceLastFetch) {
+      const keysFileUrl = await this.backendInterface.retrieveDiagnosisKeys(period);
```

Comment:
It looks like the old code wrapped this in a try catch - the danger here might be that a server or network error goes unnoticed

Reply:
Good catch!